tools/c7n_azure/tests/azure_common.py

```diff
@@ -301,7 +301,14 @@ def _replace_subscription_id(s):
                 r"[\da-zA-Z]{8}-([\da-zA-Z]{4}-){3}[\da-zA-Z]{12}" \
                 % '|'.join(['(%s)' % p for p in prefixes])
 
-        return re.sub(regex, r"\g<prefix>" + DEFAULT_SUBSCRIPTION_ID, s)
+        match = re.search(regex, s)
+        s = re.sub(regex, r"\g<prefix>" + DEFAULT_SUBSCRIPTION_ID, s)
```

Comment:
nit: I wonder if `re.sub` is the same as `s.replace(re.search(regex, s), NEW)` anyway.  e.g. can we replace this line with `s = s.replace(match, r"\g<prefix>" + DEFAULT_SUBSCRIPTION_ID)` to avoid doing the regex search twice.  The method runs a very large number of times so might have some measurable impact.